fe/fe-core/src/main/java/org/apache/doris/catalog/TableIf.java

```diff
@@ -173,5 +175,17 @@ public String toMysqlType() {
             }
         }
     }
+
+    default List<Column> getColumns() {
+        return Collections.emptyList();
+    }
+
+    default Set<String> getPartitionNames() {
+        return Collections.emptySet();
+    }
+
+    default Partition getPartition(String name) {
+        return null;
+    }
```

Comment:
i don't think Partition is a common enough struct of partition for all type of table. PTAL @morningman

Reply:
I think so and that's why it has a default implementation